Dispatcher/UI/Homepage/HomepageViewController.swift

```diff
@@ -38,58 +39,27 @@ class HomepageViewController: UIViewController, LoadingViewDelegate, UITableView
         
         tableView.dataSource = dataSource
         tableView.delegate = self
-        tableView.register(UINib(nibName: Constants.NibNames.homepage, bundle: nil), forCellReuseIdentifier: Constants.TableCellsIdentifier.homepage)
-    }
-    
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        navigationController?.isNavigationBarHidden = true
     }
 
-    func viewWillDisppear(_ animated: Bool) {
-        super.viewWillDisappear(animated)
-        navigationController?.isNavigationBarHidden = false
-    }
-    
     func fetchInitialResults() {
-        
         DispatchQueue.main.async {
             self.loadingView.isHidden = false
             self.loadingView.loadIndicator.startAnimating()
         }
-        
-        self.fetchNewsFromAPI() {
+        viewModel.fetchNewsFromAPI() {
             
             DispatchQueue.main.async {
+                self.dataSource.models = self.viewModel.newsArray
```

Comment:
I Don't think you should set the data source again since you already looking on this array as data source in line 29

Reply:
It doesn't work without it
@abenartz